File: python/leetcode/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def dfs(temp, idx):
            res.append(temp.copy())
            for i in range(idx, len(nums)):
                temp.append(nums[i])
                dfs(temp, i+1)
                temp.pop()

        dfs([], 0)
        return res

File: python/leetcode/78-subsets.cs
public class Solution
{
  public IList<IList<int>> Subsets(int[] nums)
  {
    var res = new List<IList<int>>();

    void Backtrack(List<int> temp, int i)
    {
      res.Add(new List<int>(temp));
      for (int index = i; index < nums.Length; index++)
      {
        temp.Add(nums[index]);
        Backtrack(temp, index + 1);
        temp.RemoveAt(temp.Count - 1);
      }
    }

    Backtrack(new List<int>() { }, 0);
    return res;
  }
}